source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -2295,6 +2315,16 @@ See `"not master" and "node is recovering"`_,
 and the test scenario called
 "parsing 'not master' and 'node is recovering' errors".
 
+Why close connections when server is shutting down?
+'''''''''''''''''''''''''''''''''''''''''''''''''''
+
+When a server shuts down, it will return one of the "server shutting down"
+errors for each attempted operation and eventually will close all connections.
+Keeping a connection to a server which is shutting down open would only
+produce errors on this connection - such a connection will never be usable for
+any operations. In contrast, when a server returns "not master" error the
+connection may be usable for other operations (such as secondary reads).
```

Comment:
But add a note that that is 4.2 specific.

Reply:
Okay will add this.